halo/halo.py

```diff
@@ -248,6 +248,49 @@ def animation(self, animation):
         self._animation = animation
         self._text = self._get_text(self._text['original'])
 
+    def _check_stream(self):
+        """Returns whether the stream is open, and if applicable, writable
+        Returns
+        -------
+        bool
+            Whether the stream is open
+        """
+        if self._stream.closed:
+            return False
+
+        try:
+            # Attribute access kept separate from invocation, to avoid
+            # swallowing AttributeErrors from the call which should bubble up.
+            check_stream_writable = self._stream.writable
+        except AttributeError:
+            pass
```

Comment:
Shouldn't this return `False`?

Reply:
It's some duck type checking — a stream need only have a `closed` attribute and a `write()` method; there's no guarantee it'll have a `writable()` method to determine whether it can be written to. If it doesn't have that writable-checking feature, this `pass` assumes "if it's not closed, it's writable".